server/odc-service/src/main/java/com/oceanbase/odc/service/connection/logicaldatabase/core/executor/sql/SqlExecutionCallback.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.connection.logicaldatabase.core.executor.sql;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.jdbc.core.StatementCallback;
+
+import com.oceanbase.odc.core.session.ConnectionSession;
+import com.oceanbase.odc.core.session.ConnectionSessionConstants;
+import com.oceanbase.odc.core.session.ConnectionSessionUtil;
+import com.oceanbase.odc.core.shared.Verify;
+import com.oceanbase.odc.core.sql.execute.model.JdbcGeneralResult;
+import com.oceanbase.odc.core.sql.execute.model.SqlExecuteStatus;
+import com.oceanbase.odc.core.sql.execute.model.SqlTuple;
+import com.oceanbase.odc.service.connection.logicaldatabase.core.executor.execution.model.ExecutionCallback;
+import com.oceanbase.odc.service.connection.logicaldatabase.core.executor.execution.model.ExecutionGroupContext;
+import com.oceanbase.odc.service.connection.logicaldatabase.core.executor.execution.model.ExecutionResult;
+import com.oceanbase.odc.service.connection.logicaldatabase.core.executor.execution.model.ExecutionStatus;
+import com.oceanbase.odc.service.connection.model.ConnectionConfig;
+import com.oceanbase.odc.service.connection.util.ConnectionInfoUtil;
+import com.oceanbase.odc.service.session.OdcStatementCallBack;
+import com.oceanbase.odc.service.session.factory.DruidDataSourceFactory;
+import com.oceanbase.odc.service.session.model.SqlExecuteResult;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @Author: Lebie
+ * @Date: 2024/8/28 21:04
+ * @Description: []
+ */
+@Slf4j
+public class SqlExecutionCallback implements ExecutionCallback<SqlExecuteReq, SqlExecutionResultWrapper> {
```

Comment:
In `LogicalDatabaseChangeTask#doClose`, it would release all connection sessions because they were created in the `LogicalDatabaseChangeTask#doStart`. It follows 'Whoever creates it is responsible for releasing it.'.

Reply:
I changed the logic. Now the connection session will be created right before executing, and be expired after execution.